pkg/template/api/validation/validation.go

```diff
@@ -12,6 +12,26 @@ import (
 
 var parameterNameExp = regexp.MustCompile(`^[a-zA-Z0-9\_]+$`)
 
+// totalParametersSizeLimitB is the total size limit for the template
+// parameters names and descriptions.
+// TODO: This constant was copied from k8s totalAnnotationSizeLimitB, we
+//       should make the upstream public with a better name and re-use it
+const totalParametersSizeLimitB int = 64 * 1024 // 64 kB
+
+// ValidateParametersSize validates the total size of all parameters names,
+// descriptions.
+func ValidateParametersSize(params []api.Parameter) fielderrors.ValidationErrorList {
+	allErrs := fielderrors.ValidationErrorList{}
+	var totalSize int64
+	for i := range params {
+		totalSize += (int64)(len(params[i].Name)) + (int64)(len(params[i].Description))
```

Comment:
Storage size is all we care about.  We should have a separate validation path for process (i believe) that should not be so restrictive.

----- Original Message -----
> > @@ -12,6 +12,26 @@ import (
> >  
> >  var parameterNameExp = regexp.MustCompile(`^[a-zA-Z0-9\_]+$`)
> >  
> > +// totalParametersSizeLimitB is the total size limit for the template
> > +// parameters names and descriptions.
> > +// TODO: This constant was copied from k8s totalAnnotationSizeLimitB, we
> > +//       should make the upstream public with a better name and re-use it
> > +const totalParametersSizeLimitB int = 64 * 1024 // 64 kB
> > +
> > +// ValidateParametersSize validates the total size of all parameters
> > names,
> > +// descriptions.
> > +func ValidateParametersSize(params []api.Parameter)
> > fielderrors.ValidationErrorList {
> > +	allErrs := fielderrors.ValidationErrorList{}
> > +	var totalSize int64
> > +	for i := range params {
> > +		totalSize += (int64)(len(params[i].Name)) +
> > (int64)(len(params[i].Description))
> 
> I thought we said we cared about value length when the template was going to
> be stored, and we didn't care when it was just going to be processed.
> @smarterclayton, what was the reason for validating the total size again?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2691/files#r31666672